src/vs/workbench/api/common/extHostTask.ts

```diff
@@ -279,7 +279,7 @@ export namespace TaskDTO {
 			isBackground: value.isBackground,
 			group: TaskGroupDTO.from(value.group as vscode.TaskGroup),
 			presentationOptions: TaskPresentationOptionsDTO.from(value.presentationOptions),
-			problemMatchers: value.problemMatchers,
+			problemMatchers: value.problemMatchers && Array.isArray(value.problemMatchers) ? value.problemMatchers : value.problemMatchers ? [value.problemMatchers] : [],
```

Comment:
Could use a helper here: `value.problemMatchers ? asArray(value.problemMatchers) : []`

Reply:
even better, could be `problemMatchers: asArray(value.problemMatchers`